client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -1670,6 +1670,11 @@ public ReduceFileGroups updateFileGroup(int shuffleId, int partitionId)
       throws CelebornIOException {
     Tuple2<ReduceFileGroups, String> fileGroupTuple =
         reduceFileGroupsMap.computeIfAbsent(shuffleId, (id) -> loadFileGroupInternal(shuffleId));
+    // when the cache is invalid, it should be fetched by oneself.
+    if (fileGroupTuple._2 != null) {
+      reduceFileGroupsMap.remove(shuffleId);
+      fileGroupTuple = loadFileGroupInternal(shuffleId);
```

Comment:
Should remove the invalid file group and place the correct one into reduceFileGroupsMap.

Reply:
Just delete lines https://github.com/apache/celeborn/blob/4d4b8cef47f3678a8cccedc48b6852141be3b61c/client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java#L1679-L1684

Reply:
done